### PR TITLE
[DAAS-37] Integration testing on Users

### PR DESCRIPTION
**Pull request by christiandleonr**
- Fixed issue on one friendship query
- Add integration testing for the user endpoints

### CODE REVIEW

File: src/main/java/com/easysplit/ess/user/infrastructure/persistence/validators/UserDatabaseValidator.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ private void validateUserEmailUniqueness(String email) {
         if (userEntity != null) {
             infrastructureHelper.throwIllegalArgumentException(
                     ErrorKeys.CREATE_USER_ILLEGALARGUMENT_TITLE,
-                    ErrorKeys.CREATE_USER_NOT_UNIQUE_USERNAME_MESSAGE,
+                    ErrorKeys.CREATE_USER_NOT_UNIQUE_EMAIL_MESSAGE,
                     new Object[]{ email }
```
Comment (Zastha) [inline src/main/java/com/easysplit/ess/user/infrastructure/persistence/validators/UserDatabaseValidator.java:63]: In both validate uniqueness for phone and e-mail you used black spaces and beginning and end = { email }, but in all other exceptions we have been using no black spaces = {email}

File: src/main/java/com/easysplit/ess/user/domain/models/User.java [hunk 2]
```diff
@@ -18,6 +18,10 @@ public class User {
     private Timestamp createdDate;
     private List<Link> links;
 
+    public User() {
```
Comment (Zastha) [inline src/main/java/com/easysplit/ess/user/domain/models/User.java:21]: If a constructor is empty, would it not be better to keep its body at a minimum? example: public User() {}

File: src/main/java/com/easysplit/shared/domain/models/ErrorResponse.java [hunk 3]
```diff
@@ -8,6 +8,10 @@ public class ErrorResponse  {
     private String errorMessage;
     private Throwable cause;
 
+    public ErrorResponse() {
```
Comment (Zastha) [inline src/main/java/com/easysplit/shared/domain/models/ErrorResponse.java:11]: Same comment as the other empty constructor

File: src/test/java/com/easysplit/ess/shared/utils/TestUtils.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Class with utility methods for testing
+ */
+public class TestUtils {
+    /**
+     * Private constructor to prevent instantiation of the class.
+     * This class should only hold string constants and should not be instantiated
+     */
+    private TestUtils() {
+
```
Comment (Zastha) [inline src/test/java/com/easysplit/ess/shared/utils/TestUtils.java:15]: If a constructor is not used, should we keep this empty space as a body?

File: src/main/java/com/easysplit/ess/user/infrastructure/persistence/validators/UserDatabaseValidator.java (hunk 1 above)
Comment (christiandleonr) [reply src/main/java/com/easysplit/ess/user/infrastructure/persistence/validators/UserDatabaseValidator.java]: I just made the change to keep cosistency

File: src/test/java/com/easysplit/ess/shared/utils/TestUtils.java (hunk 4 above)
Comment (christiandleonr) [reply src/test/java/com/easysplit/ess/shared/utils/TestUtils.java]: In general, for this and the above comments, let's keep the empty constructors in only one line. I made the change for all current empty constructors

**Review by Zastha — APPROVED** [review]
All Seems good

**Pull request merged**